PPOSolver.py

```diff
@@ -1,61 +1,164 @@
 from itertools import product
 from typing import List, Tuple
-
 import numpy as np
 
+from City import City, Neighborhood
+from Coordinate import Coordinate
 from Direction import Direction
+from PPO.Agent import Agent
 from Reporter import Reporter
 from Solver import Solver
 
+NEIGHBORHOOD_N = 3
+NEIGHBORHOOD_M = 3
+# If learning is too slow: Try increasing the batch size or the number of epochs.
+# If learning is unstable: Try decreasing the batch size or the number of epochs.
+# If you're facing memory constraints: Decrease the batch size.
+BATCH_SIZE = 128
+NUM_OF_EPOCHS = 5
+NUM_OF_SIMULATIONS = 6
+MAX_ITERATIONS = 100
+
 
 class PPOSolver(Solver):
     def __init__(self, n: int, m: int, t: int, reporter: Reporter):
         super().__init__(n, m, t, reporter)
-        self.all_actions = []
+        self.all_actions = self.init_all_actions()
+        self.agent = self.init_agent()
 
-        self.init_all_actions()
-
-    def init_all_actions(self):
-        possible_combinations = list(product([Direction.HORIZONTAL, Direction.VERTICAL], repeat=9))
-        self.all_actions = [np.array(combination) for combination in possible_combinations]
+    def init_agent(self) -> Agent:
+        return Agent(n_actions=len(self.all_actions),
+                     batch_size=BATCH_SIZE,
+                     n_epoch=NUM_OF_EPOCHS,
+                     input_dims=self.get_input_dims())
 
-    def solve(self) -> np.ndarray:
-        pass
+    def get_input_dims(self) -> Tuple[int]:
+        return NEIGHBORHOOD_M * NEIGHBORHOOD_M,
 
-    def train(self) -> None:
-        pass
+    def init_all_actions(self) -> List[np.ndarray]:
+        possible_combinations = list(product([Direction.HORIZONTAL, Direction.VERTICAL], repeat=9))
```

Comment:
let's replace the magic number here to be 
```suggestion
        possible_combinations = list(product([Direction.HORIZONTAL, Direction.VERTICAL], repeat=(self.get_input_dims()))
```